src/backends/perf_pt/collect.c

```diff
@@ -171,8 +171,23 @@ read_aux(void *aux_buf, struct perf_event_mmap_page *hdr,
         new_data_size = (size - tail) + head;
     }
 
-    // For now we simply crash out if we exhaust the buffer.
-    assert(trace->len + new_data_size <= trace->capacity);
+    // Decide if we have space for the new data, and reallocate the trace
```

Comment:
I think this comment is a bit on the obvious side. If we keep it, let's at least chop it down to 1 line.